Sources/AuthFoundation/Utilities/AsyncUtilities.swift

```diff
@@ -12,6 +12,34 @@
 
 import Foundation
 
+/// Shared TaskLocal data used within AuthFoundation tasks.
+///
+/// This is primarily used by unit tests to ensure the consistent passage of shared framework objects to isolate tests.
+@_documentation(visibility: internal)
+public enum TaskData {
+    /// The NotificationCenter instance that should be used when posting or observing notifications.
+    @TaskLocal public static var notificationCenter: NotificationCenter = .default
```

Comment:
What is the intention of @TaskLocal here?

Reply:
https://developer.apple.com/documentation/swift/tasklocal

It's part of Swift Concurrency that allows for contextual state to be inherited across a series of related tasks.  It's the foundation for SwiftUI's `environment` and `environmentObject` features.

Instead of using NotificationCenter's singleton when posting messages, this allows the SDK, within a task, to share state across its various calls.  This is used within unit tests to ensure that parallel tests don't clobber each other, causing false-negatives.  During regular operation, and unless explicitly overridden within a Task, its value will be the `NotificationCenter.default` singleton.